sklearn/utils/_namedarray.py

```diff
@@ -0,0 +1,103 @@
+# -*- coding: utf-8 -*-
+# Authors: Adrin Jalali <adrin.jalali@gmail.com>
+#
+# License: BSD 3 clause
+
+import numpy as np
+
+# NDArrayOperatorsMixin was added in numpy 1.13
+# TODO: cleanup once we support numpy 1.13+
+try:
+    from numpy.lib.mixins import NDArrayOperatorsMixin
+except ImportError:
+    raise NotImplementedError("In order to use NamedAraay, please upgrade your "
+                              "numpy to 1.13+!")
+
+from .validation import check_array, column_or_1d
+
+
+class NamedArray(NDArrayOperatorsMixin):
```

Comment:
If you're doing method forwarding with`__getattr__`, I'm not sure that you need to inherit from `NDArrayOperatorsMixin` at all.

Reply:
Not really, if I don't inherit from that, `x + 1` gives:

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-eaf7b6991020> in <module>
----> 1 x + 1

TypeError: unsupported operand type(s) for +: 'NamedArray' and 'int'
```